core/go/internal/flushwriter/flushwriter.go

```diff
@@ -197,7 +197,16 @@ func (w *writer[T, R]) queue(ctx context.Context, value T, flush bool) *op[T, R]
 }
 
 func (w *writer[T, R]) worker(i int) {
-	defer close(w.workersDone[i])
+	defer func() {
+		if w.workersDone[i] != nil {
```

Comment:
Why would it be closed already? Does this go routine now own the channel? What's the channel design here?